lib/iris/fileformats/netcdf.py

```diff
@@ -1356,6 +1381,109 @@ def _create_cf_dimensions(
                     size = self._existing_dim[dim_name]
                 self._dataset.createDimension(dim_name, size)
 
+    def _add_mesh(self, cube):
+        """
+        Add the cube's mesh, and all related variables to the dataset.
+        Includes all the mesh-element coordinate and connectivity variables.
+
+        ..note::
+
+            Here, we do *not* add the relevant referencing attributes to the
+            data-variable, because we want to create the data-variable later.
+
+        Args:
+
+        * cube (:class:`iris.cube.Cube`):
+            A :class:`iris.cube.Cube` to be saved to a netCDF file.
+
+        Returns:
+            * cf_mesh_name (string or None):
+            The name of the mesh variable created, or None if the cube does not
+            have a mesh.
+
+        """
+        cf_mesh_name = None
+        mesh = cube.mesh
+        if mesh:
+            if mesh in self._name_coord_map.coords:
+                cf_mesh_name = self._name_coord_map.name(mesh)
```

Comment:
@pp-mo I just need some clarity here...

If the `mesh` is in the `self._name_coord_map`, then it must have already been created (`_create_mesh`) for it to be in the map... therefore surely we don't want to do anything further, right?

Reply:
Yes, it's just like the regular dim-coords that `self._name_coord_map` was originally used for.
The logic is :  if you've seen it before, it was already created : so lookup the name that it was already given